pennylane/resource/__init__.py

```diff
@@ -11,9 +11,86 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-The ``resource`` module provides functionality to estimate the number of non-Clifford gates and
-logical qubits required to implement advanced quantum algorithms.
+r"""
+.. currentmodule:: pennylane
+
+The ``resource`` module provides classes and functionality to estimate the quantum resources
+(number of qubits, circuit depth, etc.) required to implement advanced quantum algorithms.
+
+
+Expectation Value Functions
+---------------------------
+
+.. autosummary::
+    :toctree: api
+
+    ~resource.estimate_error
+    ~resource.estimate_shots
+
+Quantum Phase Estimation Resources
+----------------------------------
+
+.. autosummary::
+    :toctree: api
+
+    ~resource.FirstQuantization
+    ~resource.DoubleFactorization
+
+Resource Classes
+----------------
+
+.. autosummary::
+    :toctree: api
+
+    ~resource.Resources
+    ~resource.ResourcesOperation

```

Comment:
These are all duplicated in the docs page. I suggest either not adding these categorisation or if we add, do it very carefully. Also having an example for `Tracker` and not for anything else could be misleading because the reader might think that this is the only way to do resource estimation.